Client/Client.cs

```diff
@@ -48,13 +48,13 @@ public class Client
         /// <param name="logger">Logger to be used for logging output/debug.</param>
         /// <param name="authToken"></param>
         public Client(
-            IEnumerable scopes,
+            IEnumerable<string> scopes,
             string appPrincipalId,
             ILogger logger)
         {
             this.ReadConfiguration();
             this.scopes = scopes;
-            this.appPrincipalId = appPrincipalId;
+            this.clientId = appPrincipalId;
```

Comment:
in term of authentication, there is the notion of appId, or clientId, but appPrincipalId does not exist
I'll do other proposal for the config file as well (that needs to change there too)